app/src/main/java/ee/ut/photomanipulation/PictureActivity.kt

```diff
@@ -0,0 +1,28 @@
+package ee.ut.photomanipulation
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.os.Bundle
+import android.os.PersistableBundle
+import android.provider.MediaStore
+import android.util.Log
+import android.widget.ImageView
+import androidx.appcompat.app.AppCompatActivity
+import java.io.File
+
+class PictureActivity : AppCompatActivity() {
+    val TAG = "pictureActivity"
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_picture)
+
+        val imagePath = intent.getStringExtra("imagePath")
+        val imageUri = Uri.fromFile(File(imagePath))
+
+        val bitmap = BitmapFactory.decodeFile(imagePath)
```

Comment:
loading bitmap straight to the memory is not recommended. We can improve it later but here are details https://developer.android.com/topic/performance/graphics/load-bitmap

Reply:
Noted. I'll merge for now to keep us going. Will put todo on the board to keep this in mind